src/distances.jl

```diff
@@ -24,64 +84,45 @@ julia> Ellipsoidal([1.0,0.5], [π/2])
 julia> Ellipsoidal([1.0,0.5,0.5], [π/2,0.0,0.0])
 ```
 """
-struct Ellipsoidal{N,T} <: Metric
-  dist::Mahalanobis{T}
 
-  function Ellipsoidal{N,T}(semiaxes, angles) where {N,T}
+struct Ellipsoidal{N,M<:Mahalanobis} <: Metric
+  dist::M
+
+  function Ellipsoidal{N,T}(semiaxes, angles; convention=:TaitBryanExtr) where {N,T}
```

Comment:
@rmcaixeta could you please replace the T by M here as well?

Reply:
Would need some extra changes. `one(M)` and `zero(M)` does not work.  Is this ok?

```julia
function Ellipsoidal{N,M}(semiaxes, angles; convention=:TaitBryanExtr) where {N,M}
    T = eltype(semiaxes)
```

Or do I replace the following `one(T)` and `zero(T)` for `1.0` and `0.0`? Λ may not end as a matrix of Floats, but `P` will be StaticMatrix{Float64} (even `cos(zero(Int))` return Float64) and Q=P*Λ*P' will also be a float

Reply:
Regarding the `cos` and `sin`, I think we should watch for the case with single precision `cos(1.0f0)`, that works correctly. The case with `Int` is less important in applications, and we can live with the return type `Float64`.